src/components/AddActivityForm.js

```diff
@@ -62,19 +62,31 @@ export default function AddActivityForm({ setActivity }) {
             )}
           </ErrorContainer>
         </StyledLabel>
-        <StyledAddButton
-          disabled={
-            newActivity.name.length > 3 || newActivity.details.length >= 500
-              ? false
-              : true
-          }
-        >
-          Add
-        </StyledAddButton>
+        <StyledAddButton disabled={disableSubmit()}>Add</StyledAddButton>
       </StyledForm>
     </>
   )
 
+  function disableSubmit() {
+    const nameTooShort = newActivity.name.length < 2
+    const nameTooLong = newActivity.name.length >= 40
+    const detailsTooLong = newActivity.details.length >= 500
+
+    if (nameTooShort) {
+      return true
+    } else if (nameTooLong) {
+      return true
+    } else if (detailsTooLong) {
+      return true
+    } else if (nameTooShort && detailsTooLong) {
+      return true
+    } else if (nameTooLong && detailsTooLong) {
+      return true
+    } else {
+      return false
+    }
+  }
+
```

Comment:
Die Funktion "disableSubmit()" kannst du glaub ich kürzer zusammenfassen. Wenn ich das richtig sehe, dann sind ein paar if-Bedingungen überflüssig. Wenn z.B. "nameTooShort" = true ist, geht er in die untere else if (nameTooShort && detailsTooLong) gar nicht mehr rein. Man könnte es auch umdrehen und den Bereich abfragen, bei dem "false" zurückgegeben wird. 

Vorschlag:

const nameLength = newActivity.name.length
const detailsLength = newActivity.details.length

if ( 2 > nameLength <= 40 && detailsLength >= 500 ) { false } 

Hier prüfst du den Bereich zwischen 2 - 40 und ob die details länger als 500 sind. Dann wird false zurückgegeben und in allen anderen Fällen true. 

Falls details, auch auf jeden Fall ausgefüllt sein sollen, dann könntest noch den Bereich begrenzen mit >0.

Könntest du mal ausprobieren, ob das so klappen würde. 
Für Fragen, steh ich Dir gern zur Verfügung.

Reply:
Wow - auf diese Schreibweise mit den logcial operators bin ich nicht gekommen - richtig gut! Vielen Dank!!! Jetzt kann ich sogar wieder einen ternary verwenden.

Reply:
Oh, ich habe aber gerade eine Warnung dazu bekommen. Vielleicht ändere ich es dann doch lieber wieder um...